app/models/repo.rb

```diff
@@ -149,6 +149,22 @@ def git_fetch
     end
   end
 
+  def create_branch(branch_name)
+    b = branches.new(
+      :name => branch_name,
+      :commit_uri => "https://github.com/#{branch_name}/commit/$commit",
+    )
+
+    # Make sure the branch is a real git branch before continuing and saving a record
+    return if !b.git_service.send(:rugged_repo).branches.exists?("origin/#{b.name}")
```

Comment:
I think there's a branch_exists? method on one of the git services (or if not, we should add it as I needed it for something else, if I recall)